packages/api-openfin/src/window.ts

```diff
@@ -524,6 +520,52 @@ class Window implements ssf.Window {
     wrappedWin.id = win.uuid + ':' + win.name;
     return wrappedWin;
   }
+
+  static getById(id: string) {
+    const appExists = (uuid): Promise<boolean> => {
+      return new Promise<boolean>(resolve => {
+        fin.desktop.System.getAllApplications((apps) => {
+          resolve(apps.filter((app) => app.uuid === uuid).length === 1);
+        });
+      });
+    };
+
+    const idRegex = '(' +    // Start group 1
+            '[\w\d-]+' +     // At least 1 word character, digit or dash
+            ')' +            // End group 1
+            ':' +            // colon
+            '\\1';            // Same string that was matched in group 1
+
+    let app = null;
+    let existsPromise: Promise<void>;
+    if (id.match(new RegExp(idRegex))) {
```

Comment:
I think this could be improved by having a branch to determine the app id, then then separately doing the appExists -> wrap code (i.e. not having to duplicate that code).